02/app.js

```diff
@@ -1,7 +1,52 @@
-
 /* rozwiązanie z pętlą for */
-const x = 4;
+// const x = prompt('Podaj liczbę!')
+
+// if (x > 0 && x <= 9) {
+// 	for (let i = 1; i <= 9; i++) {
+// 		console.log(`${x} * ${i} = ${x * i}`)
+// 	}
+// } else {
+// 	alert('podaj liczbę od 1 do 9')
+// 	location.reload()
+// }
+
+/* rozwiązanie z pętlą while, ogólnie działa ale wyświetla o jeden znak mnożenia za dużo  */
+
+function repeatStringNumTimes(num1, times) {
+	let userNumber1 = prompt(`Podaj podstawę`, num1)
+	let userNumber2 = prompt(`Podaj wykładnik`, times)
+	let repeatedString = ''
+	let result = userNumber1 ** userNumber2
+
+	while (userNumber2 > 0) {
+		repeatedString += userNumber1 + ' * '
+		userNumber2--
+	}
+
+	console.log(`${repeatedString} = ${result}`)
+}
+
+repeatStringNumTimes()
+
+/* rozwiązanie z if, niestety nie wiem dlaczego nie działa */
+
+// function repeatStringNumTimes(num1, times) {
+// 	const userNumber1 = prompt(`Podaj podstawę`, num1)
+// 	const userNumber2 = prompt(`Podaj wykładnik`, times)
```

Comment:
Dane pobierane od użytkownika zawsze są tupu `string` - dlatego `if` kończy działanie, bo nie są to dane typu `number`. Warto zrobić ```Number(prompt(`Podaj podstawę`, num1))``` i sprawdzać czy nie jest to `NaN` `!Number.isNaN(userNumber1)`